### ISSUE TITLE
aws_rds_cluster.aurora_postgres: [WARN] Error waiting for RDS Cluster state to be "available": unexpected state 'resetting-master-credentials', wanted target 'available'

### ISSUE THREAD

**Issue by Trii** [terraform-providers/terraform-provider-aws#2781]
Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.11.1

### Affected Resource(s)
- aws_rds_cluster

### Expected Behavior
What should have happened?

### Actual Behavior
```
* aws_rds_cluster.aurora_postgres: [WARN] Error waiting for RDS Cluster state to be "available": unexpected state 'resetting-master-credentials', wanted target 'available'. last error: %!s(<nil>)
```

### Steps to Reproduce
Create an Aurora Cluster using a snapshot from a compatible RDS instance
e.g.
```
  snapshot_identifier:               "" => "arn:aws:rds:us-east-1:<my account>:snapshot:mydb-dev-05-26"
```

### Important Factoids
This does not happen when creating a fresh cluster, only when using `snapshot_identifier`. It looks like the Pending states for `aws_rds_cluster`
Possible ref:
- https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_rds_cluster.go#L343
- https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_rds_cluster.go#L500

### References
https://github.com/hashicorp/terraform/pull/4490

**Comment by bflad** [comment]
Thank you for the bug report! Indeed you're looking at the right code which should be updated.

**Comment by Trii** [comment]
@bflad I was checking the AWS documentation and I couldn't find a reference for available states of a cluster, only for DB instances.

**Comment by bflad** [comment]
I have been poking around the RDS documentation as well with the same luck. I don't see anything in the SDK in terms of constants either. 🙁  In #2409 I moved the aws_db_instance pending states into variables near each other so at least we could compare and contrast. Might be useful in the aws_rds_cluster context (e.g. given `"preparing-data-migration"` existing in some but not all of these pending state lists).

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
